src/pfaffian.jl

```diff
@@ -182,18 +183,62 @@ function _logabspfaffian!(A::SkewHermTridiagonal{<:Real})
     end
     return logpf, sgn
 end
+
+function _logabspfaffian!(A::AbstractMatrix{<:Complex})
+    n = size(A, 1)
+    isodd(n) && return convert(real(eltype(A)), -Inf), zero(eltype(A))
+    tau = Array{eltype(A)}(undef, n - 2)
+    logpf = zero(real(eltype(A)))
+    phase = zero(real(eltype(A)))
+    @inbounds for k in 1:2:n-1
+        tauk = @view tau[k:end]
+
+        # Pivot if neccessary
+        @views kp = k + argmax(Iterators.map(abs, A[k+1:end, k]))
+        if kp != k + 1
+            @inbounds @simd for l in k:n
+                A[k+1,l], A[kp,l] = A[kp,l], A[k+1,l]
+            end
+            @inbounds @simd for l in k:n
+                A[l,k+1], A[l,kp] = A[l,kp], A[l,k+1]
+            end
+            phase += π
+        end
+
+        # Apply Gauss transformation and update `pf`
+        @inbounds if A[k+1,k] != zero(eltype(A))
```

Comment:
Since you have `@inbounds` on the `if` statement, it should apply to the whole block — you shouldn't need an `@inbounds` annotation on every line.

(For that matter, better to just apply `@inbounds` to the entire loop.)